demos/generomak_solps_from_raw_files.py

```diff
@@ -0,0 +1,234 @@
+
+# Copyright 2014-2017 United Kingdom Atomic Energy Authority
+#
+# Licensed under the EUPL, Version 1.1 or – as soon they will be approved by the
+# European Commission - subsequent versions of the EUPL (the "Licence");
+# You may not use this work except in compliance with the Licence.
+# You may obtain a copy of the Licence at:
+#
+# https://joinup.ec.europa.eu/software/page/eupl5
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the Licence for the specific language governing permissions and limitations
+# under the Licence.
+
+import sys
+import os
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+from scipy import linalg
+
+from raysect.core.math import Vector3D, translate, rotate_basis
+from raysect.optical import World
+from raysect.optical.observer import PinholeCamera
+from raysect.optical.material import AbsorbingSurface
+from raysect.primitive import Cylinder
+
+from cherab.core.atomic.elements import hydrogen, carbon
+from cherab.core.math.samplers import sample2d_grid, sample3d, sample3d_grid, samplevector3d_grid
+from cherab.core.model import TotalRadiatedPower
+from cherab.openadas import OpenADAS
+from cherab.solps import load_solps_from_raw_output
+
+
+###############################################################################
+# Load the simulation.
+###############################################################################
+demos_directory = os.path.dirname(__file__)
+simulation_directory = os.path.join(demos_directory, 'generomak_raw_files')
+try:
+    print('Loading simulation...')
+    sim = load_solps_from_raw_output(simulation_directory)
+except RuntimeError:
+    print('ERROR: simulation files not found. Please extract the raw '
+          'files from the zip archive before running this demo:')
+    print('> unzip {0}/generomak_raw_files.zip -d {0}/generomak_raw_files'.format(demos_directory))
+    sys.exit(1)
```

Comment:
Why ask the user to unzip files manually when the script can do it automatically?
```suggestion
try:
    print('Loading simulation...')
    sim = load_solps_from_raw_output(simulation_directory)
except RuntimeError:
    print('Extracting files...')
    with zipfile.ZipFile('generomak_raw_files.zip', 'r') as zip_ref:
        zip_ref.extractall(simulation_directory)
    sim = load_solps_from_raw_output(simulation_directory)
```

Reply:
This is a good idea, zipfile is in the python standard library, so why not to use it.

What about doing the extraction in the setup.py script? The similar way as the installation of the OpenADAS repo? We then don't have to care about the files not being extracted from the archive in every demo script.

Reply:
We've had issues in the past with re-installing and re-downloading the OpenADAS files where we get some cryptic error, which put me off adding this step to setup.py. I generally prefer the demos to be "pure": i.e. the demo should not modify the surrounding file system without being asked. But I see that some demos in other packages do write output files, and this isn't too far from that.

As there are two votes for this, I'll happily make the change.

Reply:
I understand that the unpacked files are large and they will be tracked by git if upacked, but isn't it an acceptable cost to avoid all the problems we see with having archives?

Reply:
Whether the user unzip files manually, or the script does it, in both cases we have the same problem. Namely, if we update the files in the archive, this demo will use the old versions if they were unzipped previously.

Reply:
Yes, that's true. We have a similar issue with compiling the pyx files: a user must run `./dev/build.sh` to ensure the latest versions are used. But I don't imagine we'll be updating these data files very often, if ever, so perhaps a one-time hit of c.30MB added to the repository isn't the end of the world.

@CnlPepper do you have any advice for this?

Reply:
I'm with Jack on this. I think it would be better to inform the user the files are not available when they run the demo. The error message can then describe the process to unzip the data. I'd only revisit automation if we get a lot of issues raised.

Reply:
That said, 30Mb isn't a killer, you could just include the files directly in the repo. We do that for Raysect. All the demo meshes are in the repo.

Reply:
OK, for simplicity I'll include the uncompressed files and remove all logic related to decompression. That should make the demo easier to follow.